File: src/core/models/attachment.ts
// Attachment data model
export type FileType = 'image' | 'document' | 'pdf' | 'audio' | 'video' | 'other';

export interface Attachment {
  id: string;                        // 唯一标识符 (UUID)
  filename: string;                  // 原始文件名
  file_type: FileType;               // 检测到的文件类型
  file_size: number;                 // 大小(字节)
  file_path_or_base64: string;       // 本地路径或 base64 数据 URL
  thumbnail?: string;                // 缩略图路径 (用于图像)
}

/**
 * Detect file type from filename extension
 */
export function detectFileType(filename: string): FileType {
  const ext = filename.split('.').pop()?.toLowerCase();
  const typeMap: Record<string, FileType> = {
    'jpg': 'image', 'jpeg': 'image', 'png': 'image', 'gif': 'image', 'webp': 'image',
    'doc': 'document', 'docx': 'document', 'txt': 'document', 'md': 'document',
    'pdf': 'pdf',
    'mp3': 'audio', 'wav': 'audio', 'flac': 'audio', 'ogg': 'audio',
    'mp4': 'video', 'webm': 'video', 'avi': 'video', 'mov': 'video'
  };
  return typeMap[ext || ''] || 'other';
}

/**
 * Validate Attachment data
 */
export function validateAttachment(attachment: Attachment): string | null {
  if (!attachment.id || attachment.id.length === 0) {
    return 'Attachment ID is required';
  }
  if (!attachment.filename || attachment.filename.length === 0) {
    return 'Attachment filename is required';
  }
  const validTypes: FileType[] = ['image', 'document', 'pdf', 'audio', 'video', 'other'];
  if (!validTypes.includes(attachment.file_type)) {
    return 'Attachment file_type must be one of: image, document, pdf, audio, video, other';
  }
  if (attachment.file_size <= 0) {
    return 'Attachment file_size must be greater than 0';
  }
  if (attachment.file_size > 10485760) { // 10MB
    return 'Attachment file_size exceeds maximum 10MB';
  }
  if (!attachment.file_path_or_base64 || attachment.file_path_or_base64.length === 0) {
    return 'Attachment file_path_or_base64 is required';
  }
  return null;
}

File: src-tauri/src/models/attachment.rs
// Attachment data model (Rust)
use serde::{Deserialize, Serialize};
use std::path::Path;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum FileType {
    Image,
    Document,
    Pdf,
    Audio,
    Video,
    Other,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Attachment {
    pub id: String,
    pub filename: String,
    pub file_path: String,
    pub file_type: FileType,
    pub file_size: u64,
    pub created_at: String,
}

impl Attachment {
    /// Validate Attachment data
    pub fn validate(&self) -> Result<(), String> {
        if self.id.is_empty() {
            return Err("Attachment ID is required".to_string());
        }
        if self.filename.is_empty() {
            return Err("Attachment filename is required".to_string());
        }
        if self.file_path.is_empty() {
            return Err("Attachment file_path is required".to_string());
        }
        if self.file_size < 1 {
            return Err("Attachment file_size must be positive".to_string());
        }
        // Validate timestamp
        if chrono::DateTime::parse_from_rfc3339(&self.created_at).is_err() {
            return Err("Attachment created_at must be a valid ISO 8601 timestamp".to_string());
        }
        Ok(())
    }

    /// Detect file type from filename extension
    pub fn detect_file_type(filename: &str) -> FileType {
        let ext = Path::new(filename)
            .extension()
            .and_then(|s| s.to_str())
            .map(|s| s.to_lowercase())
            .unwrap_or_default();

        match ext.as_str() {
            "jpg" | "jpeg" | "png" | "gif" | "bmp" | "webp" | "svg" => FileType::Image,
            "doc" | "docx" | "txt" | "md" | "rtf" => FileType::Document,
            "pdf" => FileType::Pdf,
            "mp3" | "wav" | "ogg" | "flac" | "aac" | "m4a" => FileType::Audio,
            "mp4" | "avi" | "mov" | "mkv" | "webm" | "flv" => FileType::Video,
            _ => FileType::Other,
        }
    }
}
